### ISSUE TITLE
Terminal Width limit question

### ISSUE THREAD

**Issue by Dragoonseal** [CleverRaven/Cataclysm-DDA#11881]
(Windows version, in case it matters.)

Currently Terminal Width in the options menu is limited to a max of 242. That number is multiplied by the fontwidth setting in \config\fonts.json (default of 8) to determine the final resolution width of 1936. I suspect this means it is intended to max at 1920 but gives a little wiggle room.

Knowing that fontwidth in fonts.json is half the equation I can raise that to get the game width resolution up to 2560 or higher, but that has the side effect of stretching out the font of the sidebar and menus, defeating the purpose.

So my question is, could the seemingly arbitrary limit of 242 for Terminal width be raised to something higher? Does this break anything? It's located in the \src\options.cpp file, but I don't have the ability to recompile the game so I can't test this myself.
##

**Comment by sparr** [comment]
The reality bubble is only 132x132 squares, so making your terminal much more than 264 (with the left half being the map view) won't be useful.

**Comment by DavidKeaton** [comment]
Beat me to it :-P Was gonna say all you'd see is `#` anyway. However there are some issues with the game screen that leave "leftovers" from other UI elements that take up the space that `w_ter` doesn't.

**Comment by Dragoonseal** [comment]
It matters for tiles, you can't see as far with those.

And besides, I'd like to just be able to play at my monitor's native resolution if nothing else.

**Comment by sparr** [comment]
Oh, yeah, I guess it's kinda wonky that we ask for window size as a multiple of the font size, even if the tiles being used are bigger than the font.

**Comment by DavidKeaton** [comment]
Hmm. Good odd point.

**Comment by BevapDin** [comment]
> It matters for tiles, you can't see as far with those.

Set the font tiles to be the same size as the real tiles.

**Comment by sparr** [comment]
I think BevapDin is specifically suggesting changing the map_fontwidth and map_fontheight in config/fonts.json

**Comment by Dragoonseal** [comment]
Editing the map_ settings in fonts.json does nothing if you have tiles enabled, so I'm not sure what you're asking.

**Comment by Dragoonseal** [comment]
Alright here's some screenshots that will perhaps help as a visual aid. This is what I get on my 2560x1600 monitor, playing with the Mshock24 tileset.

Default settings in fonts.json, max Terminal Width of 242, Terminal Length of 94:
http://i.imgur.com/VlA1Nr4.jpg
That gives me a game resolution of 1936x1504. The Terminal Length setting is no problem it can go quite high, though if I set it to go all the way to 1600 it starts hiding itself behind the task bar, thus I keep it a little lower. So I'm forced to effectively play it in 4:3 mode, that's as wide as it gets, it's the most game area I can get.

Now that's not max view range, if I press z to zoom out this is max view rage:
http://i.imgur.com/0fr6S1L.png

Now if I raise fontwidth to 12 and lower Terminal Width to 213 I get a game resolution of 2556x1504:
http://i.imgur.com/vgSqdxM.jpg
I get a good bit of extra gameplay area like that, and it's nice to actually play in widescreen, but note the sidebar font is now all stretched out, as are all other menus. The resolution itself is odd too so it does not make for fullscreening it well.

If I raise fontwidth to 16 and lower Terminal Width to 160 I get a game resolution of 2560x1504:
http://i.imgur.com/deCNXmd.jpg
I can also raise Terminal Length to 100 at this point to get a game resolution of exactly 2560x1600, which means I can play it nicely in fullscreen at my native resolution. However because of the increased fontwidth the sidebar and other menus are now stretched even farther at this point, so I actually lose gameplay area, and the stretched fonts themselves are a real pain.

So there's no good solution currently, anyone with a higher resolution monitor can't make use of it and gets forced to play in weird smaller sized windows. Ideally if Terminal Width's cap could be raised or gotten rid of entirely I could simply raise it to 320 and have 2560x1600 with no font stretching issues at all.

**Comment by Kurruk007** [comment]
This actually makes me wonder if it wouldn't be beneficial to introduce some sort of client-server architecture with the game, where one of the builds would use named pipes or tcp as an interface. This in turn could introduce different options to interact with the same game. It could bring some fresh minds to the table, normally dealing with different languages/libraries. All they would ever care is the way to connect and interact (sending actions or commands for refres, etc; and receiving data). Someone might move all graphic drawing to GPU via opengl and allow for smooth zooming.

While fixing issues for widescreen users can be a simple thing, thinking about a future or game might consider more radical option.

There is a wild idea that I had the other day. Imagine playing the game at home, tiles and everything, then leaving character with some mundane tasks and at work or in traffic connect to the same game via php+js or even php+app(android/iphone) and actually doing it, still in tile view. Once back home you could start back your tile app on main widescreen - map on left side, main view on the middle, with HUD standard right. If someone wants to watch a movie, you could switch to a different app that is set for your second monitor (4:3) that only does view and HUD.

Thin client for the win ;-)

**Comment by DavidKeaton** [comment]
I can't even imagine the extra hands we'd need.

**Comment by BevapDin** [comment]
The idea was basically make the font and the tiles have the same size, so the resulting window size would match with the tiles. But never mind, I see you already tried that.

Have you tried fullscreen? It completely ignores the terminal width/height settings.

**Comment by Dragoonseal** [comment]
Yes I've tried fullscreen, it definitely respects Terminal width/height settings.

If I set Terminal Width/Length to 240/75 so it runs at 1920x1200 and then fullscreen it just changes my monitor to run at 1920x1200 and upscales/stretches the game running at 1920x1200 to fill the screen. It looks quite bad, as my monitor doesn't deal with non-native resolutions well. Not to mention it throws off the resolution and windows of my second monitor.

Interestingly I can't manage to capture a screenshot of it while fullscreened, it just captures my desktop background and such, so I can't show you exactly what it looks like. But I can assure you it's exactly what I see when running the game in Windowed mode at 1920x1200.
